app/src/main/java/com/example/sharewhatyoucanproject/DonorActivity.kt

```diff
@@ -0,0 +1,166 @@
+package com.example.sharewhatyoucanproject
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.example.sharewhatyoucanproject.R
+import android.content.Intent
+
+import com.example.sharewhatyoucanproject.HomescreenActivity
+import com.example.sharewhatyoucanproject.MainActivity
+import com.example.sharewhatyoucanproject.DonorActivity
+
+import android.widget.EditText
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.database.DatabaseReference
+import android.app.ProgressDialog
+import android.Manifest.permission
+import android.os.Build
+
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.database.FirebaseDatabase
+import android.widget.Toast
+import android.content.pm.PackageManager
+import android.annotation.TargetApi
+import android.content.DialogInterface
+import android.app.Activity
+import android.app.AlertDialog
+import android.graphics.Bitmap
+import android.provider.MediaStore
+import android.content.ContentResolver
+import android.webkit.MimeTypeMap
+import com.google.android.gms.tasks.OnSuccessListener
+import com.example.sharewhatyoucanproject.ImageUploadInfo
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.firebase.storage.OnProgressListener
+import android.location.LocationListener
+import android.location.LocationManager
+import android.net.Uri
+import androidx.core.app.ActivityCompat
+import android.os.IBinder
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+
+import com.google.firebase.storage.UploadTask
+import java.io.IOException
+import java.util.ArrayList
+
+class DonorActivity : AppCompatActivity()
+{
+
+
+    // Folder path for Firebase Storage.
+    var Storage_Path = "All_Image_Uploads/"
+
+    // Root Database Name for Firebase Database.
+    var Database_Path = "All_Image_Uploads_Database"
+
+    // Creating button.
+    var ChooseButton: Button? = null
+    var UploadButton: Button? = null
+
+
+    // Creating EditText.
+    var ImageName: EditText? = null
+    var ImageDescription: EditText? = null
+
+    // Creating ImageView.
+    var SelectImage: ImageView? = null
+
+    // Creating URI.
+    var FilePathUri: Uri? = null
+
+    // Creating StorageReference and DatabaseReference object.
+    var storageReference: StorageReference? = null
+    var databaseReference: DatabaseReference? = null
+
+    // Image request code for onActivityResult() .
+    var Image_Request_Code = 7
+    var progressDialog: ProgressDialog? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
This is another listing nitpick, but, for better readability, I would skip a line before starting to write a function.